### PR TITLE
editoast: coverage

### LINKED ISSUE #1464: Increase the code coverage of Editoast
## User

- [ ] Study officer
- [ ] Manager
- [X] Front

## Description

Increase the number of tests in Editoast in order to increase the code coverage.

## Goal

Improve the quality of the project.

## Acceptance criteria

- [ ] Increase the code coverage of Editoast to 70%

## Difficulty

- [X] easy
- [ ] normal
- [ ] difficult
- [ ] extreme

## Definition of ready

- [x] The user story is well defined
- [x] Acceptance criterias are defined
- [x] The individual responsible for validating the user story is assigned
- [x] The team should understand how to demonstrate the features
- [x] Does not depend on an missing feature
- [x] Negociable with the implementation team
- [x] Valuable to the end users
- [x] Estimable by the implementation team
- [x] As small as it can be
- [x] Testable by the implementation team

## Definition of done

- [ ] Acceptance criteria are met
- [ ] Automated tests are written against acceptance criteria
- [ ] Project deployed to the testing environment
- [ ] Story ok-ed by Product Owner
- [ ] Documentation updated

### PR DESCRIPTION
**Pull request by Tguisnet**
close #1464

### CODE REVIEW

File: editoast/src/infra_cache.rs [hunk 1]
```diff
@@ -1047,38 +1130,129 @@ pub mod tests {
...
+        let r3_path = vec![
+            ("B", 250., 500., Direction::StartToStop),
+            ("D", 0., 480., Direction::StartToStop),
+        ];
+        infra_cache.load_route(create_route_cache(
+            "R3",
+            ObjectRef {
+                obj_type: ObjectType::Detector,
+                obj_id: "D1".into(),
+            },
```
Comment (flomonster) [inline editoast/src/infra_cache.rs:1233]: If you store `d1` you can use `d1.get_ref()`

**Review by flomonster — CHANGES_REQUESTED** [review]

File: editoast/src/models/errors/mod.rs [hunk 2]
```diff
@@ -69,34 +71,43 @@ enum InfraErrorType {
 }
 
 impl InfraError {
-    fn new_invalid_reference<T: AsRef<str>>(field: T, reference: ObjectRef) -> Self {
+    fn new_invalid_reference<T: AsRef<str>>(id: String, field: T, reference: ObjectRef) -> Self {
```
Comment (flomonster) [inline editoast/src/models/errors/mod.rs]:
```suggestion
    fn new_invalid_reference<T: AsRef<str>>(id: T, field: T, reference: ObjectRef) -> Self {
```

File: editoast/src/models/errors/mod.rs [hunk 3]
```diff
@@ -23,16 +23,18 @@ use super::invalidate_chartos_layer;
...
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
 #[serde(deny_unknown_fields)]
-struct InfraError {
+pub struct InfraError {
     field: String,
     is_warning: bool,
     #[serde(flatten)]
     sub_type: InfraErrorType,
+    #[serde(skip)]
+    id: String,
```
Comment (flomonster) [inline editoast/src/models/errors/mod.rs]:
```suggestion
    obj_id: String,
```

File: editoast/src/models/errors/routes.rs [hunk 4]
```diff
@@ -185,22 +210,213 @@ pub fn generate_errors(
...
+        railjson::{Direction, ObjectRef, ObjectType},
+    };
+
+    use super::generate_errors;
+    use super::InfraError;
+
+    #[test]
+    fn invalid_ref() {
+        let mut infra_cache = create_small_infra_cache();
+        let r_error_path = vec![
```
Comment (flomonster) [inline editoast/src/models/errors/routes.rs]: `r_error_path` is a weird name

File: editoast/src/infra_cache.rs [hunk 5]
```diff
@@ -1014,71 +1135,158 @@ pub mod tests {
...
-                (
-                    "LEFT".into(),
-                    vec![create_switch_connection("BASE".into(), "LEFT".into())],
-                ),
-                (
-                    "RIGHT".into(),
-                    vec![create_switch_connection("BASE".into(), "RIGHT".into())],
-                ),
-            ]),
+            groups,
```
Comment (flomonster) [inline editoast/src/infra_cache.rs]: I don't understand this change

**Review by flomonster — CHANGES_REQUESTED** [review]

File: editoast/src/infra_cache.rs (hunk 5 above)
Comment (Tguisnet) [reply editoast/src/infra_cache.rs]: I did these changes in order to produce some wrong switch_type_point for the errors on a switch type

File: editoast/src/models/errors/routes.rs (hunk 4 above)
Comment (Tguisnet) [reply editoast/src/models/errors/routes.rs]: it was for r(oute)_error_path, but I'll change

File: editoast/src/models/errors/mod.rs (hunk 2 above)
Comment (Tguisnet) [reply editoast/src/models/errors/mod.rs]: I thought about this, but this change forces me to have the same type for the id and the field, in editoast, the IDs are Strings and fields are STR, so if I put T here I have to convert id into str, or field into string, not sure it's very useful ? maybe create a second type U: AsRef<'str'>> for the id ?

Comment (flomonster) [reply editoast/src/models/errors/mod.rs]: Indeed you have to create a second generic type.

File: editoast/src/infra_cache.rs (hunk 5 above)
Comment (flomonster) [reply editoast/src/infra_cache.rs]: What about keeping this function producing a valid switch type and modify the returned value to create errors when necessary?

**Review by flomonster — APPROVED** [review]

**Pull request merged**